engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/ModifyProcessInstanceRejectionTest.java

```diff
@@ -256,4 +256,50 @@ public void shouldRejectCommandWhenVariableScopeIsUnknown() {
                     + "instructions with a scope element id that could not be found: 'C', 'D'",
                 PROCESS_ID));
   }
+
+  @Test
+  public void shouldRejectCommandWhenVariableScopeIsNotFlowScope() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .userTask("A")
+                .userTask("B")
+                .done())
+        .deploy();
+
+    final var processInstanceKey = ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create();
+
+    RecordingExporter.processInstanceRecords(ProcessInstanceIntent.ELEMENT_ACTIVATED)
+        .withProcessInstanceKey(processInstanceKey)
+        .withElementId("A")
+        .await();
+
+    // when
+    final var rejection =
+        ENGINE
+            .processInstance()
+            .withInstanceKey(processInstanceKey)
+            .modification()
+            .activateElement("B")
+            .withVariables(PROCESS_ID, Map.of("var", "process"))
+            .withVariables("A", Map.of("var", "A"))
+            .activateElement("A")
+            .withVariables("B", Map.of("var", "B"))
+            .expectRejection()
+            .modify();
+
+    // then
+    assertThat(rejection)
+        .describedAs("Expect that variable scopes with ids 'A' and 'B' are no flow scopes")
+        .hasRejectionType(RejectionType.INVALID_ARGUMENT)
+        .hasRejectionReason(
+            String.format(
+                "Expected to modify instance of process '%s' but it contains one or more variable "
+                    + "instructions with a scope element that doesn't belong the element's flow scope: 'A', 'B'. "
```

Comment:
> It's always an element targetted for activation, so it might also be an idea to not mention the element id directly

True. Let's keep it simple for now. We can be more explicit if needed.